core/go/internal/components/publictxmgr.go

```diff
@@ -77,7 +60,14 @@ type PublicTxManager interface {
 	QueryPublicTxForTransactions(ctx context.Context, dbTX *gorm.DB, boundToTxns []uuid.UUID, jq *query.QueryJSON) (map[uuid.UUID][]*pldapi.PublicTx, error)
 	QueryPublicTxWithBindings(ctx context.Context, dbTX *gorm.DB, jq *query.QueryJSON) ([]*pldapi.PublicTxWithBinding, error)
 	GetPublicTransactionForHash(ctx context.Context, dbTX *gorm.DB, hash tktypes.Bytes32) (*pldapi.PublicTxWithBinding, error)
-	PrepareSubmissionBatch(ctx context.Context, transactions []*PublicTxSubmission) (batch PublicTxBatch, err error)
+
+	// Perform (potentially expensive) transaction level validation, such as gas estimation. Call before starting a DB transaction
+	ValidateTransaction(ctx context.Context, dbTX *gorm.DB, transaction *PublicTxSubmission) error
+	// Write a set of validated transactions to the public TX mgr database, notifying the relevant orchestrator(s) to wake, assign nonces, and start the submission process
+	WriteNewTransactions(ctx context.Context, dbTX *gorm.DB, transactions []*PublicTxSubmission) (func(), []*pldapi.PublicTx, error)
+	// Convenience function that does ValidateTransaction+WriteNewTransactions for a single Tx
+	SingleTransactionSubmit(ctx context.Context, transaction *PublicTxSubmission) (*pldapi.PublicTx, error)
+
```

Comment:
This is the significant change. The API is massively simplified here, to just a validate call (which does the `eth_estimateGas` in simple scenarios where the `gas` limit is not provided by the caller), and a batch write.

Note the `func()` returned by the `WriteNewTransactions` - this **must** be called outside of the DB transaction on success (only) to trigger processing of the newly added public transactions.